modules/starboard/handleStarboard.js

```diff
@@ -0,0 +1,63 @@
+const { embedTypeV2, formatDiscordUserName } = require('../../src/functions/helpers');
+
+module.exports = async (client, msgReaction, isReactionRemove = false) => {
+    if (!client.botReadyAt) return;
+    const msg = msgReaction.message;
+    if (!msg.guild) return;
+    if (msg.guild.id !== client.guildID) return;
+    if (msgReaction.partial) msgReaction = await msgReaction.fetch();
+
+    const starConfig = client.configurations['starboard']['config'];
+    if (!starConfig || (starConfig.emoji !== msgReaction.emoji.name && starConfig.emoji !== msgReaction.emoji.id)) return;
+    if (isNaN(starConfig.minStars)) return disableModule('starboard', localize('starboard', 'invalid-minstars', {stars: starConfig.minStars}));
+
+    const starMsg = await client.models['starboard']['StarMsg'].findOne({
+        where: {
+            msgId: msg.id
+        }
+    });
+
+    const channel = client.channels.cache.get(starConfig.channelId);
+    if (!channel) return disableModule('starboard', localize('partner-list', 'channel-not-found', {c: starConfig.channelId}));
+    if ((msg.channel.nsfw && !channel.nsfw) || starConfig.excludedChannels.includes(msg.channel.id) || starConfig.excludedRoles.some(r => msg.member.roles.cache.has(r))) return;
+
+    let reactioncount = msgReaction.count;
+    if (!starConfig.selfStar && msgReaction.users.cache.has(msg.author.id)) reactioncount--;
+
+    const starboardMsg = starMsg ? await channel.messages.fetch(starMsg.starMsg).catch(() => {}) : null;
+    if (reactioncount < starConfig.minStars) {
+        if (isReactionRemove) {
+            if (starboardMsg) starboardMsg.delete();
+            client.models['starboard']['StarMsg'].destroy({
+                where: {
+                    msgId: msg.id
+                }
+            });
+        }
+        return;
+    }
```

Comment:
While this is pretty cool, please add some kind of mechanism to protect against users spamming the star reaction. I would suggest that a user can only star up to 5 messages per hour or something like that (this also makes stars more valuable, maybe make such a limit configurable?)

Reply:
Hm I can def implement something against spamming reactions but those wouldn't be stored in a database.

I do get the idea, however this is difficult to implement with reactions - how would the user know how long they have to wait until they can react again & if their reaction counted? I think this would lead to even more reaction spam (esp. if the bot removes invalid reactions).
I'd rather implement a delay for editing the starboard message so they bot is unlikely to get ratelimited there, and hope that server admins are able to configure the module accordingly to their servers activity.
I'll just ask on the original feature board page again.

Reply:
I would suggest removing the reactions when users are out of their configured star-amount and message them. I agree that a message-spam-limit would be useful. This is not about Discord ratelimiting the bot, instead about UX on the servers (imagine having to mark a channel as read every minute because someone is spamming star on random messages).

Reply:
That's why there's the minimum star amount, this should never be set to 1.

Reply:
Sure, but there will be situations where only one star is missing. A user then can spam the star to generate messages in the starboard-channel. 
Additionally, the community might abuse this feature to star every message (because it's fun you know), that's why I think a star-limit per user per timeframe would be a good idea.